package/MDAnalysis/analysis/pca.py

```diff
@@ -0,0 +1,240 @@
+        matrix.
+    explained_variance : array, (n_components, )
+        Percentage of variance explained by each of the selected components.
+        If a subset of components is not chosen then all components are stored
+        and the sum of explained variances is equal to 1.0.
+
+    Methods
+    -------
+    fit(traj=None, select='All', start=None, stop=None, step=None)
+        Find the principal components of selected atoms in a subset of the
+        trajectory.
+    transform(traj=None, n_components=None)
+        Take the original trajectory and project it onto the principal
+        components.
+    inverse_tranform(pca_space)
+        Take a pca_space and map it back onto the trajectory used to create it.
+    """
+
+    def __init__(self, u, select='all'):
+        """
+        Parameters
+        ----------
+        u : MDAnalysis Universe
+            The universe containing the trajectory frames for Principal
+            Component Analysis.
+        """
+        self._u = u
+        self._atoms = self._u.select_atoms(select)
+        self._n_atoms = self._atoms.n_atoms
+        self._calculated = False
+
+    def fit(self, n_components=None, start=None, stop=None,
+            step=None):
+        """ Use a subset of frames from the trajectory to generate the
+            principal components.
+
+        Parameters
+        ----------
+
+        n_components : int, optional
+            The number of principal components to be saved, default saves
+            all principal components, Default: -1
+        start : int, optional
+            First frame of trajectory to use for generation
+            of covariance matrix, Default: 0
+        stop : int, optional
+            Last frame of trajectory to use for generation
+            of covariance matrix, Default: -1
+        step : int, optional
+            Step between frames of trajectory to use for generation
+            of covariance matrix, Default: 1
+
+        Return
+        ------
+        cumulated_variance: array, (n_components, )
+            The amount of variance explained by the nth component and the n-1
+            components preceding it.
+        p_components: array, (n_components, n_atoms * 3)
+
+        """
+        self._setup_frames(self._u.trajectory, start, stop, step)
+        self.start = start
+        self.stop = stop
+        self.step = step
+
+        self.n_components = n_components
+        self.run()
+        self._calculated = True
+
+        if n_components is None:
+            return self.cumulated_variance, self.p_components
+        else:
+            return (self.cumulated_variance[:n_components],
+                    self.p_components[:n_components])
+
+    def _prepare(self):
+        n_dim = self._n_atoms * 3
+        self.cov = np.zeros((n_dim, n_dim))
+        self.mean = np.zeros(n_dim,)
+        for i, ts in enumerate(self._u.trajectory[self.start:self.stop:self.step]):
```

Comment:
There are two reasons here why this is different. 

1. You know that we have to remove the mean of the data to calculate the covariance matrix. For proteins we can't simple remove the mean of the data. That will only remove the translational mean. But we also have to remove the rotations in the system. The only way to remove translation and rotation at the same time is to align structures to a reference.  So to correctly remove the mean you first have to remove the mean in the reference structure and then align to the reference.  Removing the mean in the algorithm itself should be optional.

2. Currently you are using RMSD as a metric for the DMaps. RMSD depends strongly on the alignment of two structures. There is no way of knowing for us if a user already has aligned a trajectory to get some very specific RMSD values. There are other metric that don't depend on the alignment as much, like the fraction of native contacts.  There also isn't speaking much against optionally enabling the RMSD-metric to do an alignment, but optionally!